Microsoft.Toolkit.Uwp.UI/Extensions/ListViewBase/SmoothScrollIntoView.cs

```diff
@@ -0,0 +1,198 @@
+
+                    case ItemPlacement.Top:
+                        finalXPosition = previousXOffset + additionalHorizontalOffset;
+                        finalYPosition = maxYPosition + additionalVerticalOffset;
+                        break;
+
+                    case ItemPlacement.Centre:
+                        var centreX = (listViewBaseWidth - selectorItemWidth) / 2.0;
+                        var centreY = (listViewBaseHeight - selectorItemHeight) / 2.0;
+                        finalXPosition = maxXPosition - centreX + additionalHorizontalOffset;
+                        finalYPosition = maxYPosition - centreY + additionalVerticalOffset;
+                        break;
+
+                    case ItemPlacement.Right:
+                        finalXPosition = minXPosition + additionalHorizontalOffset;
+                        finalYPosition = previousYOffset + additionalVerticalOffset;
+                        break;
+
+                    case ItemPlacement.Bottom:
+                        finalXPosition = previousXOffset + additionalHorizontalOffset;
+                        finalYPosition = minYPosition + additionalVerticalOffset;
+                        break;
+
+                    default:
+                        finalXPosition = previousXOffset + additionalHorizontalOffset;
+                        finalYPosition = previousYOffset + additionalVerticalOffset;
+                        break;
+                }
+            }
+
+            scrollViewer.ChangeView(finalXPosition, finalYPosition, zoomFactor: null, disableAnimation);
+        }
+
+        /// <summary>
+        /// Smooth scrolling the list to bring the specified data item into view
+        /// </summary>
+        /// <param name="listViewBase">List to scroll</param>
+        /// <param name="item">The data item to bring into view</param>
+        /// <param name="itemPlacement">Set the item placement after scrolling</param>
+        /// <param name="disableAnimation">Set true to disable animation</param>
+        /// <param name="scrollIfVisibile">Set true to disable scrolling when the corresponding item is in view</param>
+        /// <param name="additionalHorizontalOffset">Adds additional horizontal offset</param>
+        /// <param name="additionalVerticalOffset">Adds additional vertical offset</param>
+        /// <returns>Note: Even though this return <see cref="Task"/>, it will not wait until the scrolling completes</returns>
+        public static async Task SmoothScrollIntoViewWithItem(this ListViewBase listViewBase, object item, ItemPlacement itemPlacement = ItemPlacement.Default, bool disableAnimation = false, bool scrollIfVisibile = true, int additionalHorizontalOffset = 0, int additionalVerticalOffset = 0)
+        {
+            await SmoothScrollIntoViewWithIndex(listViewBase, listViewBase.Items.IndexOf(item), itemPlacement, disableAnimation, scrollIfVisibile, additionalHorizontalOffset, additionalVerticalOffset);
```

Comment:
Really? I heard it a bad practice like call stack won't be proper etc...

Reply:
@vgromfeld comment?

Reply:
I can't find a solution :( it would be nice if I can see how ScrollViewer is working.

Reply:
@vgromfeld you're just saying to make a direct return here? Can you confirm and maybe just make the code suggestion?

Reply:
The suggestion is to remove the async/await keywords.:

```
public static Task SmoothScrollIntoViewWithItem(this ListViewBase listViewBase, object item, ItemPlacement itemPlacement = ItemPlacement.Default, bool disableAnimation = false, bool scrollIfVisibile = true, int additionalHorizontalOffset = 0, int additionalVerticalOffset = 0)
{
    return SmoothScrollIntoViewWithIndex(listViewBase, listViewBase.Items.IndexOf(item), itemPlacement, disableAnimation, scrollIfVisibile, additionalHorizontalOffset, additionalVerticalOffset);
}
```

It saves some size on the final binary because the compiler does not have the generate the state machine behind the async/await keywords. 
It is not always recommended to do this because if the called method is throwing, you will not see all the methods in the exception stack. Here, it means that if `SmoothScrollIntoViewWithIndex` is throwing, its exception will not contains any information about the fact that it has been called by `SmoothScrollIntoViewWithItem`.